src/components/AutocompleteMnxMetabolite.vue

```diff
@@ -119,29 +129,33 @@ export default Vue.extend({
           const searchResultsInTheModel = [] as MetaNetXMetabolite[];
           const searchResultsNotInTheModel = [] as MetaNetXMetabolite[];
           response.data.forEach((metabolite: MetaNetXMetabolite) => {
-            const annotation = metabolite.annotation;
-            const metaboliteIds = [metabolite.mnx_id];
-            for (const namespace in annotation) {
-              annotation[namespace].forEach(metaboliteId =>
-                metaboliteIds.push(metaboliteId)
-              );
-            }
+            const deprecatedIds = metabolite.annotation["deprecated"] || [];
+            delete metabolite.annotation["deprecated"];
+            const metaboliteIdsMap = {
+              ...metabolite.annotation,
+              "metanetx.chemical": [metabolite.mnx_id, ...deprecatedIds]
+            };
             let isMetaboliteFound = false;
             metabolite.modelNames = [];
-            for (const modelName in this.metabolitesInModelsMap) {
-              if (
-                metaboliteIds.some(metaboliteId =>
-                  this.metabolitesInModelsMap[modelName].has(metaboliteId)
-                )
-              ) {
-                isMetaboliteFound = true;
-                metabolite.modelNames.push(modelName);
+            for (const model in this.metabolitesInModelsMap) {
+              const [modelId, modelName] = JSON.parse(model);
+              for (const namespace in metaboliteIdsMap) {
+                metaboliteIdsMap[namespace].forEach(metaboliteId => {
+                  if (this.metabolitesInModelsMap[model].has(metaboliteId)) {
+                    isMetaboliteFound = true;
+                    metabolite.modelNames.push(modelName);
```

Comment:
This appends the same modelName multiple times if multiple ids from annotations are found